### PR TITLE
chore(ThreadMemberManager): make `#fetch` only take one object parameter

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

@Jiralite pointed out this one out

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

File: typings/index.d.ts [hunk 1]
```diff
@@ -5091,6 +5095,7 @@ export interface ThreadCreateOptions<AllowedThreadType> extends StartThreadOptio
   startMessage?: MessageResolvable;
   type?: AllowedThreadType;
   invitable?: AllowedThreadType extends 'GUILD_PRIVATE_THREAD' | 12 ? boolean : never;
+  rateLimitPerUser?: number;
```
Comment (ImRodry) [inline typings/index.d.ts:5098]: Once again xD

**Review by ImRodry — COMMENTED** [review]
Personally, I don’t agree with this. Despite the member parameter being optional, the options object is only ever used if a member is provided, so it wouldn’t make sense to do fetch({force:true}) or even fetch({cache:true}) as that can simply be replaced by fetch(true). Either way, just my two cents

**Comment by suneettipirneni** [comment]
> Personally, I don’t agree with this. Despite the member parameter being optional, the options object is only ever used if a member is provided, so it wouldn’t make sense to do fetch({force:true}) or even fetch({cache:true}) as that can simply be replaced by fetch(true). Either way, just my two cents

Yeah I'm starting to agree with this, I think I'll just remove the deprecation on the boolean overload.

**Comment by iCrawl** [comment]
This needs another rebase for the new type testing PR that landed.

**Comment by SpaceEEC** [comment]
Isn't this breaking? The second parameter exists in stable (13.3.1)
https://github.com/discordjs/discord.js/blob/13.3.1/src/managers/ThreadMemberManager.js#L110-L121

**Comment by suneettipirneni** [comment]
> Isn't this breaking? The second parameter exists in stable (13.3.1)
> 
> https://github.com/discordjs/discord.js/blob/13.3.1/src/managers/ThreadMemberManager.js#L110-L121

Welp, semver: major it is

**Review by vladfrangu — CHANGES_REQUESTED** [review]
Code doesn't align with implementation

File: packages/discord.js/src/managers/ThreadManager.js [hunk 2]
```diff
@@ -105,7 +104,6 @@ class ThreadManager extends CachedManager {
     type,
     invitable,
     reason,
-    rateLimitPerUser,
```
Comment (vladfrangu) [inline packages/discord.js/src/managers/ThreadManager.js:108]: Why was this removed?

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/discord.js/src/managers/ThreadManager.js [hunk 3]
```diff
@@ -69,7 +69,6 @@ class ThreadManager extends CachedManager {
    * `GUILD_NEWS_THREAD`</warn>
    * @property {boolean} [invitable] Whether non-moderators can add other non-moderators to the thread
    * <info>Can only be set when type will be `GUILD_PRIVATE_THREAD`</info>
-   * @property {number} [rateLimitPerUser] The rate limit per user (slowmode) for the new channel in seconds
```
Comment (almeidx) [inline packages/discord.js/src/managers/ThreadManager.js:72]: This should probably be added back too

**Review by vladfrangu — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**